Actions/CheckForUpdates/CheckForUpdates.HelperFunctions.ps1

```diff
@@ -560,3 +561,296 @@ function UpdateSettingsFile {
+- perProject: A boolean indicating whether the files are per project (default is false).
+
+.PARAMETER projects
+An array of project names used when resolving per-project file paths.
+
+.OUTPUTS
+An array of hashtables, each containing:
+- sourceFullPath: The full path to the source file.
+- originalSourceFullPath: The full path to the original source file (if found, otherwise $null).
+- type: The type of the file.
+- destinationFullPath: The full path to the destination file.
+#>
+function ResolveFilePaths {
+    Param(
+        [Parameter(Mandatory=$true)]
+        [string] $sourceFolder,
+        [string] $originalSourceFolder = $null,
+        [Parameter(Mandatory=$true)]
+        [string] $destinationFolder,
+        [array] $files = @(),
+        [string[]] $projects = @()
+    )
+
+    if(-not $files) {
+        return @()
+    }
+
+    $fullFilePaths = @()
+    foreach($file in $files) {
+        if($file.Keys -notcontains 'sourceFolder') {
+            $file.sourceFolder = '' # Default to current folder
+        }
+
+        if($file.Keys -notcontains 'filter') {
+            $file.filter = '' # Default to all files
+        }
+
+        if($file.Keys -notcontains 'origin') {
+            $file.origin = 'template' # Default to template
+        }
+
+        if($file.Keys -notcontains 'type') {
+            $file.type = '' # Default to empty
+        }
+
+        if($file.Keys -notcontains 'destinationFolder') {
+            # If destinationFolder is not specified, use the sourceFolder, so that the file structure is preserved
+            $file.destinationFolder = $file.sourceFolder
+        }
+
+        if($file.Keys -notcontains 'perProject') {
+            $file.perProject = $false # Default to false
+        }
+
+        # If originalSourceFolder is not specified, it means there is no custom template, so skip custom template files
+        if(!$originalSourceFolder -and $file.origin -eq 'custom template') {
+            Write-Host "Skipping custom template file(s) with source folder '$($file.sourceFolder)' as there is no original source folder specified"
+            continue;
+        }
+
+        # All files are relative to the template folder
+        Write-Host "Resolving files for source folder '$($file.sourceFolder)' and filter '$($file.filter)'"
+        $sourceFiles = @(Get-ChildItem -Path (Join-Path $sourceFolder $file.sourceFolder) -Filter $file.filter -File -ErrorAction SilentlyContinue | Select-Object -ExpandProperty FullName)
+
+        Write-Host "Found $($sourceFiles.Count) files for filter '$($file.filter)' in folder '$($file.sourceFolder)' (relative to folder '$sourceFolder', origin '$($file.origin)')"
+
+        if(-not $sourceFiles) {
+            Write-Debug "No files found for filter '$($file.filter)' in folder '$($file.sourceFolder)' (relative to folder '$sourceFolder')"
+            continue

```

Comment:
If the `filter` is not a wildcard, would it make sense to write a warning instead?